src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionLiteral.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed partial class LocalRewriter
+    {
+        public override BoundNode? VisitCollectionLiteralExpression(BoundCollectionLiteralExpression node)
+        {
+            Debug.Assert(!_inExpressionLambda);
+            Debug.Assert(node.Type is { });
+
+            var syntax = node.Syntax;
+            var collectionType = node.Type;
+            var constructor = node.Constructor;
+
+            BoundExpression rewrittenReceiver = constructor.IsDefaultValueTypeConstructor()
```

Comment:
oh intersting.  does this exist even for structs without a cosntructor?

Reply:
Yes, the compiler generates a `MethodSymbol` for the default parameterless struct constructor.

Reply:
what if hte struct has a non-parameterless-construct, and no default declared?  I thought (but could be entirely wrong) that we don't have a constructor in that case.  Also, for some cases from metadata.

Reply:
There is always a symbol for the parameterless constructor for struct types.